configs/defs/elasticsearch.textproto

```diff
@@ -0,0 +1,733 @@
+        "to potential attacks from untrusted networks. This helps protect sensitive data, prevent unauthorized "
+        "access, and minimize the impact of security vulnerabilities."
+      remediation:
+        "To remediate this issue, follow these steps:\n"
+        "1. Open the Elasticsearch configuration file `elasticsearch.yml`.\n"
+        "   - The file is typically located in the Elasticsearch home folder or the `config` subdirectory.\n"
+        "2. Search for the `network.host` setting in the configuration file.\n"
+        "3. Uncomment the line if present and set it to the specific network interface or IP address you want Elasticsearch to listen on.\n"
+        "   - Example: `network.host: 192.168.1.100`\n"
+        "4. If the `network.host` setting is not present, add the following line to the configuration file:\n"
+        "      network.host: <Interface_or_IP>\n"
+        "5. Save the configuration file.\n"
+        "6. Restart the Elasticsearch service for the changes to take effect.\n"
+      cis_benchmark: {
+        profile_level: 1
+        severity: LOW
+      }
+      scan_instructions:
+        # /usr/share/elasticsearch
+        "generic{check_alternatives:{"
+        "  file_checks:{"
+        "    files_to_check:{single_file:{path:\"/usr/share/elasticsearch/elasticsearch.yml\"}}"
+        "    existence:{should_exist: true}"
+        "    non_compliance_msg: \"Elasticsearch configuration file does not exist\""
+        "  }"
+        "  file_checks:{"
+        "    files_to_check:{single_file:{path:\"/usr/share/elasticsearch/elasticsearch.yml\"}}"
+        "    content_entry: {"
+        "      match_type: NONE_MATCH"
+        "      match_criteria: {"
+        "        filter_regex: \"network.host:.*\""
+        "        expected_regex: \"network.host:[\\\\s]*0\\\\.0\\\\.0\\\\.0\""
+        "      }"
+        "    }"
+        "  }"
+        "}"
+  
+        # /etc/elasticsearch
+        "check_alternatives:{"
+        "  file_checks:{"
+        "    files_to_check:{single_file:{path:\"/etc/elasticsearch/elasticsearch.yml\"}}"
+        "    existence:{should_exist: true}"
+        "    non_compliance_msg: \"Elasticsearch configuration file does not exist\""
+        "  }"
+        "  file_checks:{"
+        "    files_to_check:{single_file:{path:\"/etc/elasticsearch/elasticsearch.yml\"}}"
+        "    content_entry: {"
+        "      match_type: NONE_MATCH"
+        "      match_criteria: {"
+        "        filter_regex: \"network.host:.*\""
+        "        expected_regex: \"network.host:[\\\\s]*0\\\\.0\\\\.0\\\\.0\""
+        "      }"
+        "    }"
+        "  }"
+        "}}"
+    }
+  }
+
+  # Some rules need a workaround because the configuration value is not on the same line of the key.
+# Therefore, a workaround is needed. Using the \0 delimiter does not seem to execute the regex on all file content
```

Comment:
Let's remove the part about \0 delimiters not working (I'll look into fixing that) and just say that we split the delimiter on "true".